crates/sui-kvstore/src/writer.rs

```diff
@@ -139,64 +135,81 @@ where
                 .get_full_checkpoint_contents(&checkpoint_summary.content_digest)
                 .map_err(|_| anyhow!("Failed to read checkpoint content from store"))?
             {
-                let mut transactions = vec![];
-                let mut effects = vec![];
-                let mut events = vec![];
-                let mut transactions_to_checkpoint = vec![];
+                // let mut transactions = vec![];
+                // let mut effects = vec![];
+                // let mut events = vec![];
+                // let mut transactions_to_checkpoint = vec![];
+                let mut objects = vec![];
 
                 for content in contents.iter() {
-                    let transaction_digest = content.transaction.digest().into_inner().to_vec();
-                    effects.push((transaction_digest.clone(), content.effects.clone()));
-                    transactions_to_checkpoint.push((
-                        transaction_digest.clone(),
-                        checkpoint_summary.sequence_number,
-                    ));
-                    transactions.push((transaction_digest, content.transaction.clone()));
+                    // let transaction_digest = content.transaction.digest().into_inner().to_vec();
+                    // effects.push((transaction_digest.clone(), content.effects.clone()));
+                    // transactions_to_checkpoint.push((
+                    //     transaction_digest.clone(),
+                    //     checkpoint_summary.sequence_number,
+                    // ));
+                    // transactions.push((transaction_digest, content.transaction.clone()));
 
-                    if let Some(event_digest) = content.effects.events_digest() {
-                        if let Some(tx_events) = store
-                            .get_transaction_events(event_digest)
-                            .map_err(|_| anyhow!("Failed to fetch events from the store"))?
-                        {
-                            events.push((event_digest.into_inner().to_vec(), tx_events));
-                        }
+                    // if let Some(event_digest) = content.effects.events_digest() {
+                    //     if let Some(tx_events) = store
+                    //         .get_transaction_events(event_digest)
+                    //         .map_err(|_| anyhow!("Failed to fetch events from the store"))?
+                    //     {
+                    //         events.push((event_digest.into_inner().to_vec(), tx_events));
+                    //     }
+                    // }
+                    let object_keys: Vec<_> = content
+                        .effects
+                        .all_changed_objects()
```

Comment:
@mystenmark: double checking if `all_changed_objects` is the right collection to use here for backfill

Reply:
I think this is correct - cc @lxfind

Reply:
If the goal is to write all the output objects, this is correct.